File: week8/parabs.cpp
#include <iostream>
using namespace std;
// Abstract class
class Shape {
   protected:
    float dimension;
   public:
    void getDimension() {
        cin >> dimension;
    }
    void display(){
        cout<<"I AM PARTIAL"<<endl;
    }
    // pure virtual Function
    virtual float calculateArea() = 0;
};
class Square : public Shape {
   public:
    float calculateArea() {
        return dimension * dimension;
    }
};
class Circle : public Shape {
   public:
    float calculateArea() {
        return 3.14 * dimension * dimension;
    }
};
int main() {
    Square square;
    Circle circle;
    square.display();
    cout << "Enter the length of the square: "<<endl;
    square.getDimension();
    circle.display();
    cout << "Area of square: " << square.calculateArea() << endl;
    cout << "Enter radius of the circle: "<<endl;
    circle.getDimension();
    cout << "Area of circle: " << circle.calculateArea() << endl;
}

File: week8/parabs.java
abstract class Shape
{
    abstract float area();
    void display()
    {
        System.out.println("This is a shape");
    }
}
class Rectangle extends Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
}
class Triangle extends Shape
{
    float a,b;
    Triangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.50*a*b;
    }
}
public class PartialAbstraction
{
    public static void main(String[] args) 
    {
        Rectangle obj=new Rectangle(23,45);
        Triangle obj1=new Triangle(67,89);
        System.out.println("Area of Rectangle : "+obj.area());
        System.out.println("Area of Triangle : "+obj1.area());
    }
}
